src/MinGW.cmake

```diff
@@ -102,7 +118,11 @@ macro(configure_mingw_vcpkg_after)
     include("${ProjectOptions_SRC_DIR}/Utilities.cmake")
     detect_architecture(_arch)
     string(TOLOWER "${_arch}" _arch)
-    set(Z_VCPKG_TARGET_TRIPLET_ARCH ${_arch})
+    if(CROSSCOMPILING AND TARGET_ARCHITECTURE)
```

Comment:
Does this condition work? From my experience of working with CMake's confusing datatypes, an explicit `"${variable}" STREQL ""` works better for making sure that the condition is correct.